src/RateChangeQueue.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+library RateChangeQueue {
+    struct RateChange {
+        uint256 rate;
+        uint256 untilEpoch;
+    }
+
+    struct Queue {
+        // Map from index to RateChange
+        mapping(uint256 => RateChange) changes;
+        uint256 head;
+        uint256 tail;
+    }
+
+    function enqueue(Queue storage queue, uint256 rate, uint256 untilEpoch) internal {
```

Comment:
TODO: drop enqueued rate-changes if `queue.changes[queue.tail].untilEpoch == untilEpoch`.

Reply:
@Stebalien I want to keep the queue implementation simpler. I've made this change at the call site in `modifyRailPayment`.